libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/AppLockView/SFSDKAppLockViewController.m

```diff
@@ -0,0 +1,201 @@
+        if (currentPasscode) {
+            [SFSecurityLockout setUserDeclinedBiometricUnlock:YES];
+            SFSecurityLockoutAction action = isVerificationMode?SFSecurityLockoutActionPasscodeVerified:SFSecurityLockoutActionPasscodeCreated;
+            [self unlock:currentPasscode lockAction:action];
+        } else {
+            [self dismissStandaloneBiometricSetup];
+        }
+    }
+}
+
+- (void)dismissStandaloneBiometricSetup
+{
+    [SFSecurityLockout setupTimer];
+    [[[SFSDKWindowManager sharedManager] passcodeWindow].viewController dismissViewControllerAnimated:NO completion:^{
+        [[SFSDKWindowManager sharedManager].passcodeWindow dismissWindowAnimated:NO withCompletion:^{}];
+    }];
+}
+
+-(UIViewController *)controllerFromMode:(SFAppLockControllerMode) mode configData:(SFAppLockConfigurationData)configData andViewConfig:(SFSDKAppLockViewConfig *)viewConfig
+{
+    UIViewController *currentViewController = nil;
+    
+    if (mode == SFBiometricControllerModeEnable || mode == SFBiometricControllerModeVerify) {
+        SFSDKBiometricViewController *bvc = [[SFSDKBiometricViewController alloc] initWithAppLockConfigData:configData viewConfig:viewConfig];
+        bvc.biometricResponseDelgate = self;
+        bvc.verificationMode = (mode == SFBiometricControllerModeVerify);
+        currentViewController = bvc;
+    } else if (mode == SFPasscodeControllerModeCreate || mode == SFPasscodeControllerModeChange) {
+        SFSDKPasscodeCreateController *pvc = [[SFSDKPasscodeCreateController alloc] initWithPasscodeConfigData:configData viewConfig:viewConfig];
+        pvc.createDelegate = self;
+        pvc.updateMode = (mode == SFPasscodeControllerModeChange);
+        currentViewController = pvc;
+    } else {
+        SFSDKPasscodeVerifyController *pvc = [[SFSDKPasscodeVerifyController alloc] initWithPasscodeConfigData:configData viewConfig:viewConfig];
+        pvc.verifyDelegate = self;
+        currentViewController = pvc;
+    }
+    
+    return currentViewController;
+}
+
+#pragma mark - private methods
+- (BOOL)canPromptBiometricEnrollment
+{
+    return !([SFSecurityLockout userDeclinedBiometricUnlock] || [SFSecurityLockout biometricUnlockEnabled]) && [self canShowBiometric];
+}
+
+- (BOOL)canShowBiometric
+{
+    return [SFSecurityLockout biometricUnlockAllowed] && [[SFPasscodeManager sharedManager] deviceHasBiometric];
+}
+
+- (void)promptBiometricEnrollment:(NSString *)passcode
+{
+    SFSDKBiometricViewController *pvc = [[SFSDKBiometricViewController alloc] initWithAppLockConfigData:self.configData viewConfig:self.viewConfig];
+    pvc.biometricResponseDelgate = self;
+    pvc.currentPasscode = passcode;
+    [self pushViewController:pvc animated:NO];
+}
+
+- (void)unlock:(NSString *)passcode lockAction:(SFSecurityLockoutAction) action{
+    [self.navigationController popViewControllerAnimated:NO];
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [SFSecurityLockout setupTimer];
```

Comment:
We should consider encapsulating the logic here in form of a delegate callback to SFSecurityLockout.